pkg/java/pom/artifact.go

```diff
@@ -105,9 +108,28 @@ func evaluateVariable(s string, props map[string]string) string {
 		} else {
 			// <properties> might include another property.
 			// e.g. <animal.sniffer.skip>${skipTests}</animal.sniffer.skip>
-			newValue = evaluateVariable(props[m[1]], props)
+			ss, ok := props[m[1]]
+			if ok {
+				// search for looped properties
+				if slices.Contains(usedProps, ss) {
+					printLoopedPropertiesStack(m[0], usedProps)
+					return ""
+				}
+				usedProps = append(usedProps, ss) // save received props to check if we get this prop again
+				newValue = evaluateVariable(ss, props, usedProps)
+				usedProps = []string{} // clear props if we returned from recursive. Required for correct work with 2 same props like ${foo}-${foo}
+			}
+
 		}
 		s = strings.ReplaceAll(s, m[0], newValue)
 	}
 	return s
 }
+
+func printLoopedPropertiesStack(env string, usedProps []string) {
+	var s string
+	for _, prop := range usedProps {
+		s += fmt.Sprintf("%s -> ", prop)
+	}
+	log.Logger.Debugf("Unable to get variable. Lopped Properties was find: %s%s", s, env)
```

Comment:
I think it doesn't happen usually. Can be a warning.

```suggestion
	log.Logger.Warnf("Lopped properties were detected: %s%s", s, env)
```

Reply:
Agree with you, thanks!